crates/forge-runner/src/gas.rs

```diff
@@ -126,47 +126,13 @@ fn get_messages_costs(
     starknet_gas_usage + sharp_gas_usage
 }
 
-fn clear_compiled_class_hash_update(
-    state_changes: blockifier::state::cached_state::StateChanges,
-) -> blockifier::state::cached_state::StateChanges {
-    // copy-pasted blockifier::state::cached_state::StateChanges
-    // link: https://github.com/starkware-libs/blockifier/blob/eb4958ad98d92dc8f8b493edc8dce1a79038c94d/crates/blockifier/src/state/cached_state.rs#L319
-    struct StateMapsCopy {
-        _nonces: HashMap<ContractAddress, Nonce>,
-        _class_hashes: HashMap<ContractAddress, ClassHash>,
-        _storage: HashMap<StorageEntry, Felt252>,
-        compiled_class_hashes: HashMap<ClassHash, CompiledClassHash>,
-        _declared_contracts: HashMap<ClassHash, bool>,
-    }
-
-    // use to cast blockifier::state::cached_state::StateChanges into same struct but public for us
-    union StateMapsHack {
-        origin: std::mem::ManuallyDrop<blockifier::state::cached_state::StateChanges>,
-        public: std::mem::ManuallyDrop<StateMapsCopy>,
-    }
-
-    let mut u = StateMapsHack {
-        origin: std::mem::ManuallyDrop::new(state_changes),
-    };
-    unsafe {
-        // compiled_class_hash_updates is used only for keeping track of declares
-        // which we don't want to include in gas cost
-        // clear on public struct
-        (*u.public).compiled_class_hashes.clear();
```

Comment:
Let's not use this hack again 😅. This only works if the layout of the struct remains the same, and Rust does not guarantee this. The compiler could change the layout to optimize the code, leading to a really difficult bug to find.

The patch is the same alternative to code